src/Controller/CRUDController.php

```diff
@@ -901,8 +927,13 @@ final public function configureAdmin(Request $request): void
             ));
         }
 
+        $pool = $this->get('sonata.admin.pool');
+        \assert($pool instanceof Pool);
+
         try {
-            $this->admin = $this->get('sonata.admin.pool')->getAdminByAdminCode($adminCode);
+            /** @phpstan-var AdminInterface<T> $admin */
+            $admin = $pool->getAdminByAdminCode($adminCode);
```

Comment:
maybe we could replace the try catch wit hasAdminByCode if? not sure if the pool has that method

Reply:
We can let it that way it will be soon changed: https://github.com/sonata-project/SonataAdminBundle/pull/7190